libnetwork/network.go

```diff
@@ -1955,12 +1958,20 @@ func (n *Network) hasLoadBalancerEndpoint() bool {
 	return len(n.loadBalancerIP) != 0
 }
 
-func (n *Network) ResolveName(req string, ipType int) ([]net.IP, bool) {
+func (n *Network) ResolveName(ctx context.Context, req string, ipType int) ([]net.IP, bool) {
 	var ipv6Miss bool
 
 	c := n.getController()
 	networkID := n.ID()
+
+	ctx, span := otel.Tracer("").Start(ctx, "Network.ResolveName", trace.WithAttributes(
+		attribute.String("network-name", n.Name()),
+		attribute.String("network-id", networkID),
```

Comment:
We probably need to define a proper naming schema for these attributes. For the record, the OTEL sdk adds these attributes to all spans:

* `internal.span.format`
* `otel.library.name`
* `span.kind`

So I think those attributes should be named something like:

* `network-name` -> `libnet.network.name`
* `network-id` -> `libnet.network.id`